src/block.rs

```diff
@@ -1,34 +1,118 @@
 //use std::stream::Stream;
 //use futures::io::AsyncRead;
 
-#[derive(Debug)]
+use serde::{Serialize, Deserialize};
+
+use crate::burnfee::BurnFee;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
 pub struct Block {
-    
+    pub body:     BlockBody,
+    pub is_valid: u8,
```

Comment:
Just for context, because I have no idea if we want to keep this --- JS uses is_valid as a cheap form of error-catching. Not sure if we still need it in RUST.

The case I'd be concerned about is -- if we create a block from transactions in our mempool and it is invalid because one of the TXS is invalid, do we want to tank all of the transactions in our mempool just because the block is bad?

is_valid is JS-poor-man's way of saying, "just mark the bad ones and we can clean-up later".

Not sure how parallelization works in Rust. It might be useful to keep depend on what happens there, as tagging the block as invalid and then doing a sanity-check at the end when all of the children have finished processing might be cleaner than noping out halfway through. Not sure.

That was the point in JS anyway.